### DISCUSSION TITLE
Expected performance for 'Splicing Cargo workspace"

### DISCUSSION THREAD

**Discussion by rbtcollins** [bazelbuild/rules_rust#2259] [General]
```
Fetching repository @crate_index; Splicing Cargo workspace. 400s
```

is not uncommon for a fairly modest Bazel monorepo, which has 2 Rust projects in it (and many more other things). Is this usual, and if not, how do I get the splicer to provide a profile / opentelemetry trace / debug logging?

**Comment by UebelAndre** [comment]
In the next release, setting `CARGO_BAZEL_DEBUG` should enable some tracing to show you what's happening in splicing. In general though this time is spent having cargo fetch a copy of the registry and then all crates you depend on. Setting `CARGO_BAZEL_ISOLATED=false` should allow crate universe to use the standard cargo location on your machine to speed things up.

**Reply by rbtcollins** [reply]
Does that isolated setting cache the registry in some manner that permits incremental updates? This could explain it somewhat.

**Reply by rbtcollins** [reply]
Tested on a faster machine.
Splicing with the current config ~250s, stays about that when repeated
with CARGO_BAZEL_ISOLATED=false  ~ 100s
repeating the isolated run a second time - ~20s

**Comment by erenon** [comment]
Try reducing the number of supported platforms, by specifying `supported_platform_triples`:

```python
crate.from_specs(
  supported_platform_triples = [
    "x86_64-unknown-linux-gnu",
    "aarch64-apple-darwin",
  ],
)
```
